### PR TITLE
Implement Codegen for `ErrorNode`

### PR DESCRIPTION
**Pull request by kirawoods**

### CODE REVIEW

File: ee/codegen/src/__test__/helpers/node-data-factories.ts [hunk 1]
```diff
@@ -954,6 +955,32 @@ export function apiNodeFactory({
...
+      label: "Error Node",
+      name: "error-node",
+      targetHandleId: "06573a05-e6f0-48b9-bc6e-07e06d0bc1b1",
+      errorSourceInputId,
+      errorOutputId: "69250713-617d-42a4-9326-456c70d0ef20",
+    },
+    inputs: [
+      {
+        id: errorSourceInputId,
+        key: "error_source",
```
Comment (NgoHarrison) [inline ee/codegen/src/__test__/helpers/node-data-factories.ts]: nit: Going off the real data model, maybe the key should be named `error_source_input_id`

File: ee/codegen/src/__test__/helpers/node-data-factories.ts [hunk 2]
```diff
@@ -954,6 +955,32 @@ export function apiNodeFactory({
...
+  const errorSourceInputId = "d2287fee-98fb-421c-9464-e54d8f70f046";
+
+  return {
+    id: "2cd960a3-cb8a-43ed-9e3f-f003fc480951",
+    type: "ERROR",
+    data: {
+      label: "Error Node",
+      name: "error-node",
+      targetHandleId: "06573a05-e6f0-48b9-bc6e-07e06d0bc1b1",
+      errorSourceInputId,
```
Comment (NgoHarrison) [inline ee/codegen/src/__test__/helpers/node-data-factories.ts]: nit: `errorSourceInputId: errorSourceInputId,`

File: ee/codegen/src/generators/nodes/error-node.ts
Comment (NgoHarrison) [inline ee/codegen/src/generators/nodes/error-node.ts:1]: Let's include a python field for `error_output_id` as well

File: ee/codegen/src/__test__/nodes/__snapshots__/error-node.test.ts.snap [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+    node_id = UUID("2cd960a3-cb8a-43ed-9e3f-f003fc480951")
+    label = "Error Node"
+    error_output_id = UUID("69250713-617d-42a4-9326-456c70d0ef20")
+    source_handle_id = UUID("e4dedb66-0638-4f0c-9941-6420bfe353b2")
+    target_handle_id = UUID("370d712d-3369-424e-bcf7-f4da1aef3928")
+    error_inputs_by_name = {
+        "error_source_input_id": VellumError(
+            message="Something went wrong!", code="USER_DEFINED_ERROR"
+        )
+    }
```
Comment (noanflaherty) [inline ee/codegen/src/__test__/nodes/__snapshots__/error-node.test.ts.snap:23]: [blocking] I wouldn't expect this to show up here. `error_inputs_by_name` isn't an attribute of `BaseErrorNodeDisplay`

File: ee/codegen/src/generators/nodes/error-node.ts [hunk 4]
```diff
@@ -13,12 +14,79 @@ export class ErrorNode extends BaseSingleFileNode<
...
+      python.field({
+        name: "error_inputs_by_name",
+        initializer: python.TypeInstantiation.dict(
+          Array.from(this.nodeInputsByKey.entries()).map(([key, value]) => ({
+            key: python.TypeInstantiation.str(key),
+            value: value,
+          }))
+        ),
+      })
+    );
```
Comment (noanflaherty) [inline ee/codegen/src/generators/nodes/error-node.ts:88]: We can get rid of this.

File: ee/codegen_integration/fixtures/simple_error_node/code/display/nodes/error_node.py [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+
+
+class ErrorNodeDisplay(BaseErrorNodeDisplay[ErrorNode]):
+    name = "error-node"
+    node_id = UUID("e5ff9360-a29c-437b-a9c1-05fc52df2834")
+    label = "Error Node"
+    error_output_id = UUID("77839b3c-fe1c-4dcb-9c61-2fac827f729b")
+    source_handle_id = UUID("e4dedb66-0638-4f0c-9941-6420bfe353b2")
+    target_handle_id = UUID("370d712d-3369-424e-bcf7-f4da1aef3928")
+    error_inputs_by_name = {"error_source_input_id": Inputs.query}
```
Comment (noanflaherty) [inline ee/codegen_integration/fixtures/simple_error_node/code/display/nodes/error_node.py]: Again wouldn't expect `error_inputs_by_name` to show up here.

File: ee/codegen_integration/fixtures/simple_error_node/code/inputs.py [hunk 6]
```diff
@@ -0,0 +1,5 @@
+from vellum.workflows.inputs import BaseInputs
+
+
+class Inputs(BaseInputs):
+    query: str
```
Comment (noanflaherty) [inline ee/codegen_integration/fixtures/simple_error_node/code/inputs.py]: nit: We could rename this to `custom_error` for clarity.

File: ee/vellum_ee/workflows/display/nodes/vellum/error_node.py [hunk 7]
```diff
@@ -11,6 +11,8 @@
 
 class BaseErrorNodeDisplay(BaseNodeVellumDisplay[_ErrorNodeType], Generic[_ErrorNodeType]):
     error_output_id: ClassVar[Optional[UUID]] = None
+    error_inputs_by_name: ClassVar[Dict[str, Any]] = {}
```
Comment (noanflaherty) [inline ee/vellum_ee/workflows/display/nodes/vellum/error_node.py:14]: I think we remove this `error_inputs_by_name`.

File: ee/vellum_ee/workflows/display/nodes/vellum/error_node.py [hunk 8]
```diff
@@ -11,6 +11,8 @@
...
+    name: ClassVar[str] = "error-node"
```
Comment (noanflaherty) [inline ee/vellum_ee/workflows/display/nodes/vellum/error_node.py:15]: Oh wow, didn't realize error nodes have names.

File: src/vellum/workflows/nodes/displayable/bases/error_node.py [hunk 9]
```diff
@@ -0,0 +1,23 @@
+from typing import ClassVar, Generic, Union
+
+from vellum.workflows.errors.types import VellumError
+from vellum.workflows.nodes.bases import BaseNode
+from vellum.workflows.types.generics import StateType
+
+
+class BaseErrorNode(BaseNode[StateType], Generic[StateType]):
```
Comment (noanflaherty) [inline src/vellum/workflows/nodes/displayable/bases/error_node.py]: [blocking] We were going to remove this file, yeah?

File: ee/codegen/src/__test__/nodes/__snapshots__/error-node.test.ts.snap (hunk 3 above)
Comment (NgoHarrison) [reply ee/codegen/src/__test__/nodes/__snapshots__/error-node.test.ts.snap:23]: I included this in the display class to generate the inputs for the node. I couldn't find/don't know an existing way to grab the values for it

Comment (kirawoods) [reply ee/codegen/src/__test__/nodes/__snapshots__/error-node.test.ts.snap:23]: Definitely had a hard time trying to take this piece out. It looks like it's pretty integrated into how we generate node_inputs in the serializer here: https://github.com/vellum-ai/vellum-python-sdks/blob/main/ee/vellum_ee/workflows/display/nodes/vellum/error_node.py#L30

Comment (noanflaherty) [reply ee/codegen/src/__test__/nodes/__snapshots__/error-node.test.ts.snap:23]: I’ll take a pass at this.

Comment (noanflaherty) [reply ee/codegen/src/__test__/nodes/__snapshots__/error-node.test.ts.snap:23]: Actually, let’s merge this without this change and Kira, please make me a ticket in our epic for me to try to clean up at a later time.

**Review by noanflaherty — APPROVED** [review]

File: ee/codegen/src/__test__/nodes/__snapshots__/error-node.test.ts.snap (hunk 3 above)
Comment (kirawoods) [reply ee/codegen/src/__test__/nodes/__snapshots__/error-node.test.ts.snap:23]: Sounds like a plan!

File: ee/vellum_ee/workflows/display/nodes/vellum/error_node.py (hunk 8 above)
Comment (dvargasfuertes) [reply ee/vellum_ee/workflows/display/nodes/vellum/error_node.py:15]: They don't...

**Review by dvargas92495 — COMMENTED** [review]
We likely should remove `"name"`

✅

**Comment by kirawoods** [comment]
> We likely should remove `"name"`
> 
> ✅

Weird... adding this to the cleanup ticket

File: ee/vellum_ee/workflows/display/nodes/vellum/error_node.py (hunk 8 above)
Comment (noanflaherty) [reply ee/vellum_ee/workflows/display/nodes/vellum/error_node.py:15]: They actually do in the main repo for some reason. Check out `ErrorNodeData`.

**Pull request merged**